src/templates/codebuild-deployment-template.yaml

```diff
@@ -0,0 +1,1074 @@
+    Type: AWS::Lambda::Function
+    Properties:
+      Runtime: python3.13
+      Handler: index.handler
+      Role: !GetAtt rResourceCleanupRole.Arn
+      Timeout: 300
+      Code:
+        ZipFile: |
+          import boto3
+          import cfnresponse
+          def handler(event, context):
+              try:
+                  if event['RequestType'] == 'Delete':
+                      # Clean up S3 bucket
+                      s3 = boto3.client('s3')
+                      bucket = event['ResourceProperties']['BucketName']
+                      paginator = s3.get_paginator('list_object_versions')
+                      for page in paginator.paginate(Bucket=bucket):
+                          delete_keys = []
+                          if 'Versions' in page:
+                              delete_keys.extend([{'Key': obj['Key'], 'VersionId': obj['VersionId']} for obj in page['Versions']])
+                          if 'DeleteMarkers' in page:
+                              delete_keys.extend([{'Key': obj['Key'], 'VersionId': obj['VersionId']} for obj in page['DeleteMarkers']])
+                          if delete_keys:
+                              s3.delete_objects(Bucket=bucket, Delete={'Objects': delete_keys})
+
+                      # Clean up EventBridge rule
+                      events = boto3.client('events')
+                      lambda_client = boto3.client('lambda')
+                      stack_name = event['ResourceProperties']['StackName']
+                      rule_name = f"{stack_name}-pipeline-monitor"
+                      function_name = f"{stack_name}-pipeline-status-function"
+
+                      try:
+                          # Remove targets first
+                          events.remove_targets(Rule=rule_name, Ids=['1'])
+                          # Delete the rule
+                          events.delete_rule(Name=rule_name)
+                          # Remove Lambda permission
+                          lambda_client.remove_permission(
+                              FunctionName=function_name,
+                              StatementId=rule_name
+                          )
+                      except Exception as e:
+                          print(f"EventBridge cleanup error (non-fatal): {e}")
+
+                  cfnresponse.send(event, context, cfnresponse.SUCCESS, {})
+              except Exception as e:
+                  cfnresponse.send(event, context, cfnresponse.FAILED, {}, str(e))
+
+  # Create S3 bucket to store configuration files
+  rConfigBucket:
+    # checkov:skip=CKV_AWS_18:Bucket used to trigger CodePipeline, access logs are not needed
+    # checkov:skip=CKV_AWS_21:Bucket used to trigger CodePipeline, object versioning is not needed
+    Metadata:
+      cdk_nag:
+        rules_to_suppress:
+          - id: AwsSolutions-S1
+            reason: Bucket used to trigger CodePipeline, access logs are not needed
+    Type: AWS::S3::Bucket
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We recommend that you use S3 Lifecycle configuration whenever defining an S3 bucket in CloudFormation or CDK. Setting lifecycle rules for the bucket helps in cleaning up unused data from buckets which in turn leads to cost and storage savings.

[Learn more](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig.html)